src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -28,19 +28,53 @@
  * Contact : chris@qbittorrent.org
  */
 
-#include <QFile>
-#include <QHostAddress>
+#include "filterparserthread.h"
+
 #include <QDataStream>
+#include <QFile>
 #include <QStringList>
 
 #include <libtorrent/session.hpp>
 #include <libtorrent/ip_filter.hpp>
 
 #include "base/logger.h"
-#include "filterparserthread.h"
 
 namespace libt = libtorrent;
 
+namespace
+{
+    bool parseIPAddress(QString _ip, libt::address &address)
+    {
+        _ip = _ip.trimmed();
+
+        // Emule .DAT files contain leading zeroes in IPv4 addresses
+        // eg 001.009.106.186
+        // We need to remove them because both QHostAddress and Boost.Asio fail to parse them.
+        QStringList octets = _ip.split('.', QString::SkipEmptyParts);
+        if (octets.size() == 4) {
+            QString octet; // it is faster to not recreate this object in the loop
+            for (int i = 0; i < 4; i++) {
+                octet = octets[i];
```

Comment:
1. maybe faster & more bulletproof
 ```c++
QString &octet = octets[i].right(3);
```

2. for consistency use `++i`

3. then you can just
 ```c++
if (octet[0] == '0') {
    if (octet[1] == '0')
      octet.remove(0, 2);
    else
      octet.remove(0, 1);
}
```